pkg/flowaggregator/flowaggregator.go

```diff
@@ -321,46 +337,229 @@ func (fa *flowAggregator) initExportingProcess() error {
 }
 
 func (fa *flowAggregator) Run(stopCh <-chan struct{}) {
-	exportTicker := time.NewTicker(fa.exportInterval)
-	defer exportTicker.Stop()
 	go fa.collectingProcess.Start()
 	defer fa.collectingProcess.Stop()
 	go fa.aggregationProcess.Start()
 	defer fa.aggregationProcess.Stop()
+	go fa.flowRecordExpiryCheck(true, stopCh)
+	go fa.flowRecordExpiryCheck(false, stopCh)
+
+	<-stopCh
+}
+
+func (fa *flowAggregator) flowRecordExpiryCheck(isActiveTimeout bool, stopCh <-chan struct{}) {
+	var timeout time.Duration
+	var updatePQ ipfixintermediate.FlowKeyRecordMapCallBack
+	if isActiveTimeout {
+		timeout = fa.activeFlowRecordTimeout
+		updatePQ = fa.updateActiveExpiryPriorityQueue
+	} else {
+		timeout = fa.inactiveFlowRecordTimeout
+		updatePQ = fa.updateInactiveExpiryPriorityQueue
+	}
+
+	expireTicker := time.NewTicker(timeout)
 	for {
 		select {
 		case <-stopCh:
 			if fa.exportingProcess != nil {
 				fa.exportingProcess.CloseConnToCollector()
 			}
+			expireTicker.Stop()
 			return
-		case <-exportTicker.C:
+		case <-expireTicker.C:
 			if fa.exportingProcess == nil {
 				err := fa.initExportingProcess()
 				if err != nil {
-					klog.Errorf("Error when initializing exporting process: %v, will retry in %s", err, fa.exportInterval)
-					// Initializing exporting process fails, will retry in next exportInterval
+					klog.Errorf("Error when initializing exporting process: %v, will retry in %s", err, fa.activeFlowRecordTimeout)
+					// Initializing exporting process fails, will retry in next cycle.
 					continue
 				}
 			}
-			err := fa.aggregationProcess.ForAllRecordsDo(fa.sendFlowKeyRecord)
-			if err != nil {
-				klog.Errorf("Error when sending flow records: %v", err)
+			// Pop the flow record item from priority queue and send the flow records
+			if err := fa.popAndSendFlowRecords(isActiveTimeout); err != nil {
+				klog.Errorf("Error when triggering active timeout expiry actions: %v", err)
 				// If there is an error when sending flow records because of intermittent connectivity, we reset the connection
 				// to IPFIX collector and retry in the next export cycle to reinitialize the connection and send flow records.
 				fa.exportingProcess.CloseConnToCollector()
 				fa.exportingProcess = nil
+				expireTicker.Stop()
+				expireTicker = time.NewTicker(timeout)
 				continue
 			}
+
+			if err := fa.aggregationProcess.ForAllRecordsDo(updatePQ); err != nil {
+				klog.Errorf("Error when updating active timeout priority queue: %v", err)
```

Comment:
There would be no error returned for `updateActiveExpiryPriorityQueue`, maybe we can change the error info to be "error when updating **inactive** timeout priority queue"